File: WdtcCore/src/main/java/org/wdt/wdtc/core/game/config/DefaultGameConfig.java
package org.wdt.wdtc.core.game.config;

import com.google.gson.annotations.SerializedName;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.wdt.wdtc.core.game.Launcher;
import org.wdt.wdtc.core.utils.JavaUtils;

@Setter
@Getter
@ToString
public class DefaultGameConfig {
  @SerializedName("info")
  private VersionInfo info;
  @SerializedName("config")
  private Config config;

  public DefaultGameConfig(Launcher launcher) {
    this.info = launcher.getVersionInfo();
    this.config = new Config();
  }

  public DefaultGameConfig() {
  }

  @Setter
  @Getter
  @ToString
  public static class Config {
    @SerializedName("RunningMemory")
    private int Memory;
    @SerializedName("JavaPath")
    private String JavaPath;
    @SerializedName("WindowWidth")
    private int width;
    @SerializedName("WindowHeight")
    private int hight;

    public Config() {
      this.Memory = 1024;
      this.JavaPath = JavaUtils.getRunJavaHome();
      this.width = 1000;
      this.hight = 618;
    }

    public Config(int memory, String javaPath, int width, int hight) {
      Memory = memory;
      JavaPath = javaPath;
      this.width = width;
      this.hight = hight;
    }
  }


}

File: WdtcCore/src/main/kotlin/org/wdt/wdtc/core/game/config/DefaultGameConfig.kt
package org.wdt.wdtc.core.game.config

import com.google.gson.annotations.SerializedName
import org.wdt.wdtc.core.game.Version
import org.wdt.wdtc.core.utils.runJavaHome

data class DefaultGameConfig(
  @field:SerializedName("info")
  var info: VersionInfo,

  @field:SerializedName("config")
  var config: Config

) {
  constructor(version: Version) : this(version.versionInfo, Config())

  data class Config(
    var memory: Int = 1024,

    var javaPath: String = runJavaHome,

    var width: Int = 1000,

    var hight: Int = 618
  )
}
